### ISSUE TITLE
Memory leak when destroying/un-mounting an Animated component

### ISSUE THREAD

**Issue by VitorLBonfim** [software-mansion/react-native-reanimated#5280]
### Description

When an Animated React component is unmounted after an animation, the allocated memory is never released, thus increasing to a app memory usage increase.

### Video
https://github.com/software-mansion/react-native-reanimated/assets/120739994/700d3879-2cd2-4ee7-b501-5382310f76b1

### Steps to reproduce

As shown in the video above, open your memory inspector (I've used both the built-in memory inspector and also xCode) and just start tapping into the button, memory will quickly rise and never release again.

### Snack or a link to a repository

 https://github.com/vitorclelis96/ReanimatedMemoryLeak-example

### Reanimated version

3.5.4

### React Native version

0.72.6

### Platforms

iOS

### JavaScript runtime

None

### Workflow

React Native

### Architecture

None

### Build type

Release mode

### Device

iOS simulator

### Acknowledgements

Yes

**Comment by piaskowyk** [comment]
Hey 👋 We well look at it 🫡

**Comment by Jrh0203** [comment]
Are there any updates on this?

**Comment by MehediH** [comment]
@piaskowyk hey! any update on this? The issue is still present in the latest release - unmounted components aren't releasing the memory, leading to apps using high amounts of memory

**Comment by l2hyunwoo** [comment]
I think this is a very serious issue for React Native app's user experience, and it's a bit disappointing to see no response.

**Comment by alexfoxy** [comment]
Yeah, this has some serious implications for my app. I have a gallery where users can view images, and every time it is closed the view is not cleaned up causing the app to quickly run out of memory. I've tried running the component without re-animated views and the memory is freed over time.
